test/JoinAlgorithmsTest.cpp

```diff
@@ -248,25 +249,31 @@ struct RowAdderWithUndef {
   const auto& getOutput() const { return output_; }
 };
 
+// Join both vectors `a` and `b` and assert that the result is equal to the
+// given `expected` result. Joins are performed 2 times, the second time with
+// `a` and `b` swapped.
 void testDynamicJoinWithUndef(const std::vector<std::vector<FakeId>>& a,
                               const std::vector<std::vector<FakeId>>& b,
                               std::vector<std::array<FakeId, 2>> expected,
                               source_location l = source_location::current()) {
+  using namespace std::placeholders;
+  using namespace std::ranges;
   auto trace = generateLocationTrace(l);
-  auto compare = [](auto l, auto r) { return static_cast<Id>(l) < r; };
+  auto compare = [](FakeId l, FakeId r) {
+    return static_cast<Id>(l) < static_cast<Id>(r);
+  };
+  AD_CONTRACT_CHECK(is_sorted(a | views::join, compare));
```

Comment:
Even more idiomatic would be to implement a functor in `ad_utility`, s.t. you can write
` is_sorted(a | views::join, {}, ad_utility::static_cast<Id>)` without the code above:)